src/js/utils/deviceFeaturesUtils.js

```diff
@@ -106,3 +108,33 @@ export const getUnavailableCapabilities = (
     });
     return list;
 };
+
+/**
+ * In current versions of FW, TT uses hex raw bytes and T1 uses UTF-8 representation of revision.
+ * Examples:
+ * T1: df0963ec48f01f3d07ffca556e21ff0070cab099, f4424ece1ccb7fc0d6cad00ff840fac287a34f07
+ * TT: 6466303936336563, 623139636266363763
+ * To avoid being model specific, in case this problem is fixed, it is required to reliably detect what encoding is used.
+ * @param {Features} features
+ * @returns revision in UTF-8 format or null
+ */
+export const parseRevision = (features: Features) => {
+    const { revision } = features;
+
+    // if device is in bootloader mode, revision is null
+    if (!revision) return null;
+
+    // if revision contains at least one a-f character and the rest are numbers, it is in UTF-8 (no encoding needed)
+    if (/^(?=.*[a-fA-f])([a-fA-F0-9]*)$/.test(revision)) {
```

Comment:
I though that a mere letter is enough to be sure the revision is in the ut8 format. Couldn't we just test for a presence of a letter?

```suggestion
    if (/[a-fA-f]/.test(revision)) {
```

Reply:
I need to test that it consists only from numbers and a-f and it contains at least one a-f.